app/controllers/application_controller.rb

```diff
@@ -323,14 +323,21 @@ def calc_layout_params
   # having the project param stored inside the query record, q.
   def set_project_ivar
     # NOTE: Query param projects is always an array of ids.
-    query_projects = params.dig(:q, :projects) || []
-    # If more than one project, it's none. Only want single-project associations
+    query_projects = extract_query_projects
+    # If more than one project, it's none.
+    # Only want single-project associations
     query_project = query_projects.size > 1 ? nil : query_projects.first
     project_id = params[:project] || query_project
     # At this point, we still might not have one. That's fine - just return nil.
     @project = Project.safe_find(project_id)
   end
 
+  def extract_query_projects
+    query = query_from_q_param # includes fallback for old q
+    query&.params&.dig(:projects) || []
+  end
+  private :extract_query_projects
+
```

Comment:
At this point it seems like these two lines could probably go back in the `set_project_ivar` method without violating ABC, but IDK.

Reply:
Good point. I'll check it out.